InnoClinic.AuthorizationAPI/BLL/Services/AuthService.cs

```diff
@@ -1,32 +1,77 @@
-﻿using System.Globalization;
+﻿using System.Diagnostics;
 using AutoMapper;
 using BLL.DTO;
 using BLL.Exceptions;
+using BLL.Helpers;
 using BLL.Interfaces;
+using DAL.Constants;
 using DAL.Entities;
 using DAL.Interfaces;
+using Microsoft.AspNetCore.Http;
 using Microsoft.AspNetCore.Identity;
 
 namespace BLL.Services;
 
-public class AuthService(IAuthRepository authRepository, IMapper mapper) : IAuthService
+public class AuthService(IAuthRepository authRepository, IMapper mapper, IJwtTokenHelper jwtTokenHelper, IHttpContextAccessor httpContextAccessor) : IAuthService
 {
     public async Task<IdentityResult> RegisterAsync(RegisterDto dto, CancellationToken cancellationToken = default)
     {
-       if (dto.Password is null) throw new PasswordNullException();
-       return await authRepository.RegisterAsync(mapper.Map<User>(dto), dto.Password);
+        if (dto.Password is null) throw new ArgumentNullException(nameof(dto.Password));
+        if (dto.UserName is null) throw new ArgumentNullException(nameof(dto.UserName));
+        
+        var user = mapper.Map<User>(dto);
+
+        return await authRepository.RegisterAsync(user, dto.Password, cancellationToken);
     }
+
+    public async Task<string> LogInAsync(LogInDto dto, CancellationToken cancellationToken = default)
```

Comment:
can't we use only one method (`LogInAsync`) to authorize? you have many duplications because of `LogInAsAWorkerAsync` it will be better to pass required roles via DTO and check them instead of having 2 similar methods